fastlane/lib/assets/completions/completion.fish

```diff
@@ -0,0 +1,37 @@
+# This function was taken from https://github.com/Carthage/Carthage/blob/master/Source/Scripts/carthage-fish-completion
+function __fish_prog_needs_subcommand
+  set cmd (commandline -opc)
+  if [ (count $cmd) -eq 1 -a $cmd[1] = 'fastlane' ]
+    return 0
+  end
+    return 1
+end
+
+if test -e "Fastfile"
+  set file "Fastfile"
+else if test -e "fastlane/Fastfile"
+  set file "fastlane/Fastfile"
+else if test -e ".fastlane/Fastfile"
+  set file ".fastlane/Fastfile"
+else
+  exit 1
+end
+
+set commands (string match --regex '.*lane\ \:(?!private_)([^\s]*)\ do' (cat $file))
```

Comment:
Hmm, I think this is fine for now as it'll require some more thinking. I don't like that we have essentially duplicate logic across 3 files right now. 

Just rebase to the current master please, then we can merge your PR